shanoir-ng-front/src/app/shared/notifications/notifications.service.ts

```diff
@@ -14,75 +14,144 @@
+                if (Date.now() - new Date(task.lastUpdate).getTime() > this.TIMEOUT) {
+                    task.status = -1;
+                    task.message = 'timeout';
+                }
+            }
+            if (task.status == -1 && task.lastUpdate) {
+                let freshError: boolean = !!this.tasksInProgress.find(tip => task.id == tip.id) || (Date.now() - new Date(task.lastUpdate).getTime()) <= (this.readInterval + 1000);
                 if (freshError) {
-                    this.nbNewError++;
-                    this.freshCompletedTasks.push(task);
-                    // remove after 30s
-                    setTimeout(() => {
-                        this.freshCompletedTasks = this.freshCompletedTasks.filter(tip => tip.id != task.id);
-                    }, 30000);
+                    this.freshTimeouts[task.id]?.triggerNow();
+                    this.pushToFreshError(task);
                 }
             } else if (task.status == 1) {
-                let freshDone: Task = this.tasksInProgress.find(tip => tip.status == 2 && task.id == tip.id);
+                let freshDone: boolean = !!this.tasksInProgress.find(tip => task.id == tip.id) || (Date.now() - new Date(task.lastUpdate).getTime()) <= (this.readInterval + 1000);
                 if (freshDone) {
-                    this.nbNew++;
-                    this.freshCompletedTasks.push(task);
-                    // remove after 30s
-                    setTimeout(() => {
-                        this.freshCompletedTasks = this.freshCompletedTasks.filter(tip => tip.id != task.id);
-                    }, 30000);
+                    this.freshTimeouts[task.id]?.triggerNow();
+                    this.pushToFreshCompleted(task);
                 }
-            } else if (task.status == 2) {
+            } else if (task.status == 2 || task.status == 5) {
+                this.freshTimeouts[task.id]?.triggerNow();
                 tmpTasksInProgress.push(task);
+            } else if (task.status == 4) {
+                this.freshTimeouts[task.id]?.triggerNow();
+                tmpTasksInWait.push(task);
             }
         }
         this.tasksInProgress = tmpTasksInProgress;
+        this.tasksInWait = tmpTasksInWait;
+    }
+
+    pushToFreshCompleted(task: Task) {
+        if (!this.freshCompletedTasks.find(tip => tip.id == task.id)) {
+            this.freshCompletedTasks.push(task);
+        }
+        this.nbNew++;
+        // remove after 30min
+        this.freshTimeouts[task.id] = new SuperTimeout(() => {
+            this.freshCompletedTasks = this.freshCompletedTasks.filter(tip => tip.id != task.id);
+        }, this.persistenceTime);
     }
 
-    private connect() {
+    pushToFreshError(task: Task) {
+        if (!this.freshCompletedTasks.find(tip => tip.id == task.id)) {
+            this.freshCompletedTasks.push(task);
+        }
+        this.nbNewError++;
+        // remove after 30min
```

Comment:
Should we really remove  download errors after 30mn ? Can we wait for the error to be "seen" by the user ?

Reply:
it removes it only from the left side menu